2021-03-24 17:25:55,594 : INFO : root : Loading config.
2021-03-24 17:25:55,600 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2021-03-24 17:25:55,601 : INFO : ConfigProcessor : Loading config from cache: C:\Users\Noah\AppData\Local\Temp\f1e6b3d603b69cf20c882e489c97a885.mpf_cache
2021-03-24 17:25:55,749 : INFO : ConfigProcessor : Loading config from cache: C:\Users\Noah\AppData\Local\Temp\3667aa7b16c33035bf5ed76426078541.mpf_cache
2021-03-24 17:25:55,751 : INFO : ConfigProcessor : Loading config from cache: C:\Users\Noah\AppData\Local\Temp\a7eac71e9da1e9cfd4db34c71351dbda.mpf_cache
2021-03-24 17:25:55,752 : INFO : ConfigProcessor : Loading config from cache: C:\Users\Noah\AppData\Local\Temp\087a53a03e25f86845675c231fd2982a.mpf_cache
2021-03-24 17:25:55,753 : INFO : ConfigProcessor : Loading config from cache: C:\Users\Noah\AppData\Local\Temp\8252fe0f7adb2196fe883aefa2a8963e.mpf_cache
2021-03-24 17:25:55,754 : INFO : ConfigProcessor : Loading config from cache: C:\Users\Noah\AppData\Local\Temp\fab15a6988082b7b90a1ae2ab7ba49ba.mpf_cache
2021-03-24 17:25:55,755 : INFO : Machine : Mission Pinball Framework Core Engine v0.54.6
2021-03-24 17:25:55,755 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'jsonlogging': False, 'logfile': 'logs\\2021-03-24-17-25-55-mpf-DESKTOP-VL0THH1.log', 'pause': False, 'production': False, 'text_ui': False, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': None, 'syslog_address': None, 'mc_file_name': None, 'no_sound': False}
2021-03-24 17:25:55,755 : INFO : Machine : MPF path: c:\users\noah\mpfenv\lib\site-packages\mpf
2021-03-24 17:25:55,755 : INFO : Machine : Machine path: C:\Users\Noah\work\Andy Cavatorta\pinball\notes\MPF\test_platform
2021-03-24 17:25:55,756 : INFO : Machine : Platform: win32
2021-03-24 17:25:55,756 : INFO : Machine : Python executable location: c:\users\noah\mpfenv\scripts\python.exe
2021-03-24 17:25:55,756 : INFO : Machine : Python version: 3.7.9 (64-bit)
2021-03-24 17:25:55,759 : INFO : Machine : Initialise MPF.
2021-03-24 17:25:55,879 : INFO : P-Roc : Connecting to P-ROC
2021-03-24 17:25:57,300 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:25:57,301 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:25:59,698 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:25:59,698 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:02,085 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:02,085 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:04,470 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:04,470 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:06,857 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:06,857 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:09,243 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:09,243 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:11,629 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:11,629 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:14,014 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:14,014 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:16,401 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:16,401 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:18,787 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:18,787 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:21,172 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:21,172 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:23,559 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): Verify Chip ID took too long to receive data. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:23,559 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:24,559 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): FT_ListDevices(2)
. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:24,560 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:25,561 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): FT_ListDevices(2)
. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:25,561 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:26,563 : WARNING : P-Roc : Failed to instantiate pinproc.PinPROC(7): FT_ListDevices(2)
. Is your P/P3-Roc connected and powered up?
2021-03-24 17:26:26,563 : INFO : P-Roc : Will retry creating PinPROC in 1s.
2021-03-24 17:26:28,076 : INFO : P-Roc : Successfully connected to P-ROC/P3-ROC. Firmware Version: 2. Firmware Revision: 6. Hardware Board ID: 1
2021-03-24 17:26:28,076 : WARNING : P-Roc : Consider upgrading the firmware of your P/P3-Roc to at least 2.14. Your version contains known bugs.
2021-03-24 17:26:28,142 : INFO : EventManager : Event: ======'machine_var_p_roc_version'====== Args={'value': 2, 'prev_value': None, 'change': True}
2021-03-24 17:26:28,142 : INFO : EventManager : Event: ======'machine_var_p_roc_revision'====== Args={'value': 6, 'prev_value': None, 'change': True}
2021-03-24 17:26:28,142 : INFO : EventManager : Event: ======'machine_var_p_roc_hardware_version'====== Args={'value': 1, 'prev_value': None, 'change': True}
2021-03-24 17:26:28,143 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,245 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'MPF v0.54.6', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,245 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'MPF v0.54.6, Config version:5, Show version: 5, BCP version:1.1', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,245 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.7.9', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,246 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-10-10.0.19041-SP0', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,246 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,246 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '10', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,247 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.19041', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,247 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2021-03-24 17:26:28,247 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2021-03-24 17:26:28,296 : INFO : EventManager : Event: ======'drop_target_bank_drop_target_bank_1_up'====== Args={}
2021-03-24 17:26:28,301 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2021-03-24 17:26:28,328 : INFO : EventManager : Event: ======'bcp_connection_attempt'====== Args={'name': 'local_display', 'host': 'localhost', 'port': 5050}
2021-03-24 17:26:28,330 : INFO : BCPClientSocket : Connecting BCP to 'local_display' at localhost:5050...
2021-03-24 17:26:30,344 : INFO : BCPClientSocket : Connected BCP to 'local_display' localhost:5050
2021-03-24 17:26:30,344 : INFO : EventManager : Event: ======'bcp_clients_connected'====== Args={}
2021-03-24 17:26:30,345 : INFO : ball_device.bd_trough : State: idle
2021-03-24 17:26:30,356 : INFO : EventManager : Event: ======'init_phase_3'====== Args={}
2021-03-24 17:26:30,369 : INFO : EventManager : Event: ======'init_phase_4'====== Args={}
2021-03-24 17:26:30,370 : INFO : EventManager : Event: ======'machine_var_mc_version'====== Args={'value': 'MPF-MC v0.54.0', 'prev_value': None, 'change': True}
2021-03-24 17:26:30,370 : INFO : EventManager : Event: ======'machine_var_mc_extended_version'====== Args={'value': 'MPF-MC v0.54.0 (config_version=5, BCP v1.1, Requires MPF v0.54.0)', 'prev_value': None, 'change': True}
2021-03-24 17:26:30,374 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,375 : INFO : BallController : Initial balls found: 0
2021-03-24 17:26:30,376 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_left_flipper'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,376 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_right_flipper'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,377 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_left_slingshot_main'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,377 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_right_slingshot_main'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,377 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_left_slingshot_mid'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,378 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_right_slingshot_mid'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,378 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_drop_target_1'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,379 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_drop_target_2'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,379 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_pop_bumper_1'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,380 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_pop_bumper_2'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,380 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_pop_bumper_3'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,381 : INFO : EventManager : Event: ======'machine_var_audits_switches_s_trough1'====== Args={'value': 0, 'prev_value': None, 'change': True}
2021-03-24 17:26:30,382 : INFO : EventManager : Event: ======'init_phase_5'====== Args={}
2021-03-24 17:26:30,382 : INFO : EventManager : Event: ======'init_done'====== Args={}
2021-03-24 17:26:30,383 : INFO : EventManager : Event: ======'machine_reset_phase_1'====== Args={}
2021-03-24 17:26:30,396 : INFO : EventManager : Event: ======'slide_welcome_slide_removed'====== Args={'_from_bcp': True}
2021-03-24 17:26:30,400 : INFO : EventManager : Event: ======'slide_welcome_slide_created'====== Args={'_from_bcp': True}
2021-03-24 17:26:30,404 : INFO : EventManager : Event: ======'slide_welcome_slide_active'====== Args={'_from_bcp': True}
2021-03-24 17:26:30,405 : INFO : EventManager : Event: ======'machine_reset_phase_2'====== Args={}
2021-03-24 17:26:30,405 : INFO : EventManager : Event: ======'machine_reset_phase_3'====== Args={}
2021-03-24 17:26:30,405 : INFO : drop_target_bank.drop_target_bank_1 : All targets are already up. Will not reset bank.
2021-03-24 17:26:30,419 : INFO : EventManager : Event: ======'reset_complete'====== Args={}
2021-03-24 17:26:30,420 : INFO : EventManager : Event: ======'mode_attract_will_start'====== Args={}
2021-03-24 17:26:30,422 : INFO : EventManager : Event: ======'mode_attract_starting'====== Args={}
2021-03-24 17:26:30,422 : INFO : Mode.attract : Started. Priority: 10
2021-03-24 17:26:30,422 : INFO : EventManager : Event: ======'modes_active_modes_changed'====== Args={}
2021-03-24 17:26:30,422 : INFO : EventManager : Event: ======'mode_attract_started'====== Args={}
2021-03-24 17:26:30,423 : INFO : EventManager : Event: ======'collecting_balls_complete'====== Args={}
2021-03-24 17:26:30,424 : INFO : Machine : Starting the main run loop.
2021-03-24 17:26:30,433 : INFO : EventManager : Event: ======'slide_awesome_slide_created'====== Args={'_from_bcp': True}
2021-03-24 17:26:30,439 : INFO : EventManager : Event: ======'slide_awesome_slide_active'====== Args={'_from_bcp': True}
2021-03-24 17:27:30,436 : INFO : EventManager : Event: ======'slide_press_start_created'====== Args={'_from_bcp': True}
2021-03-24 17:27:30,438 : INFO : EventManager : Event: ======'slide_press_start_active'====== Args={'_from_bcp': True}
2021-03-24 17:28:30,432 : INFO : EventManager : Event: ======'slide_mission_pinball_created'====== Args={'_from_bcp': True}
2021-03-24 17:28:30,434 : INFO : EventManager : Event: ======'slide_mission_pinball_active'====== Args={'_from_bcp': True}
2021-03-24 17:29:17,728 : ERROR : asyncio : Exception in callback Util.raise_exceptions(<Task finishe...wer supply.')>) at c:\users\noah\mpfenv\lib\site-packages\mpf\core\utility_functions.py:726
handle: <Handle Util.raise_exceptions(<Task finishe...wer supply.')>) at c:\users\noah\mpfenv\lib\site-packages\mpf\core\utility_functions.py:726>
Traceback (most recent call last):
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\platforms\p_roc_common.py", line 143, in read_events_and_watchdog
    self.proc.flush()
OSError: Error in WriteData: wrote 0 of 8 bytes

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Noah\AppData\Local\Programs\Python\Python37\lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\core\utility_functions.py", line 734, in raise_exceptions
    future.result()
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\platforms\p_roc_common.py", line 302, in _poll_events
    loop=self.machine.clock.loop)
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\platforms\p_roc_common.py", line 152, in read_events_and_watchdog
    self.log.name) from error
mpf.exceptions.runtime_error.MpfRuntimeError: Runtime Error in P-Roc: Communication with P/P3-Roc broke down. Check USB cable and power supply. Error Code: RE-P-Roc-2 (https://docs.missionpinball.org/en/0.54/logs/RE-P-Roc-2.html)
2021-03-24 17:29:17,732 : INFO : Machine : Shutting down...
2021-03-24 17:29:17,732 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2021-03-24 17:29:17,748 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\platforms\p_roc_common.py", line 143, in read_events_and_watchdog
    self.proc.flush()
OSError: Error in WriteData: wrote 0 of 8 bytes

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\core\machine.py", line 745, in _run_loop
    raise self._exception['exception']
  File "C:\Users\Noah\AppData\Local\Programs\Python\Python37\lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\core\utility_functions.py", line 734, in raise_exceptions
    future.result()
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\platforms\p_roc_common.py", line 302, in _poll_events
    loop=self.machine.clock.loop)
  File "c:\users\noah\mpfenv\lib\site-packages\mpf\platforms\p_roc_common.py", line 152, in read_events_and_watchdog
    self.log.name) from error
mpf.exceptions.runtime_error.MpfRuntimeError: Runtime Error in P-Roc: Communication with P/P3-Roc broke down. Check USB cable and power supply. Error Code: RE-P-Roc-2 (https://docs.missionpinball.org/en/0.54/logs/RE-P-Roc-2.html)
2021-03-24 17:29:17,753 : INFO : root : MPF run loop ended.
